test/test_vertexmodel.cpp

```diff
@@ -0,0 +1,149 @@
+      mModel->setGeometry( mLineGeometry, QgsCoordinateReferenceSystem() );
+      QVERIFY( !mModel->canRemoveVertex() );
+      mModel->setEditingMode( VertexModel::EditVertex );
+      QVERIFY( mModel->canRemoveVertex() );
+
+      mModel->setEditingMode( VertexModel::NoEditing );
+      QCOMPARE( mModel->editingMode(), VertexModel::NoEditing );
+      mModel->setCurrentVertex( 0 );
+      QCOMPARE( mModel->editingMode(), VertexModel::EditVertex );
+
+      QCOMPARE( mModel->vertexCount(), 3 );
+      mModel->removeCurrentVertex();
+      QCOMPARE( mModel->vertexCount(), 2 );
+      QVERIFY( !mModel->canRemoveVertex() );
+
+      // polygon
+      mModel->setGeometry( mPolygonGeometry, QgsCoordinateReferenceSystem() );
+      QCOMPARE( mModel->editingMode(), VertexModel::NoEditing );
+      QVERIFY( !mModel->canRemoveVertex() );
+      mModel->setCurrentVertex( 0 );
+      mModel->removeCurrentVertex();
+      QCOMPARE( mModel->vertexCount(), 3 );
+      QVERIFY( !mModel->canRemoveVertex() );
+    }
+
+    void addVertexTest()
+    {
+      mModel->setGeometry( mPolygonGeometry, QgsCoordinateReferenceSystem() );
+      QCOMPARE( mModel->vertexCount(), 4 );
+      mModel->setEditingMode( VertexModel::AddVertex );
+      QCOMPARE( mModel->vertexCount(), 5 );
+
+      mModel->setGeometry( mLineGeometry, QgsCoordinateReferenceSystem() );
+      mModel->setEditingMode( VertexModel::AddVertex );
+      QCOMPARE( mModel->mCurrentIndex, 1 );
+      QVERIFY( mModel->canPreviousVertex() );
+      mModel->previous();
+      QVERIFY( !mModel->canPreviousVertex() );
+      QCOMPARE( mModel->mCurrentIndex, 0 );
+      mModel->next();
+      QCOMPARE( mModel->mCurrentIndex, 1 );
+
+      mModel->setGeometry( mLineGeometry, QgsCoordinateReferenceSystem() );
+      mModel->setEditingMode( VertexModel::AddVertex );
+      QCOMPARE( mModel->mCurrentIndex, 1 );
+      QCOMPARE( mModel->currentPoint(), QgsPoint( 1, 2 ) );
+      mModel->next();
+      QCOMPARE( mModel->mCurrentIndex, 2 );
+      QCOMPARE( mModel->currentPoint(), QgsPoint( 3, 3 ) );
+      mModel->next();
+      QCOMPARE( mModel->mCurrentIndex, 3 );
+      QCOMPARE( mModel->currentPoint().x(), 4.5 );
+      QCOMPARE( mModel->currentPoint().y(), 3.0 );
+      QCOMPARE( mModel->currentPoint(), QgsPoint( 4.5, 3 ) );
+      QVERIFY( !mModel->canNextVertex() );
+      mModel->next();
+      QCOMPARE( mModel->mCurrentIndex, 3 );
+      mModel->previous();
+      QCOMPARE( mModel->mCurrentIndex, 2 );
+      QCOMPARE( mModel->currentPoint(), QgsPoint( 3, 3 ) );
+      mModel->previous();
+      QCOMPARE( mModel->mCurrentIndex, 1 );
+      QCOMPARE( mModel->currentPoint().x(), 1.0 );
+    }
+
+    void transformTest()
+    {
+      MapSettings *mapSettings = new MapSettings();
+      mapSettings->setDestinationCrs( QgsCoordinateReferenceSystem::fromEpsgId( 21781 ) );
+      mModel->setMapSettings( mapSettings );
+      QCOMPARE( mModel->mapSettings()->destinationCrs().authid(), "EPSG:21781" );
```

Comment:
`QStringLiteral( )` and the error should be gone

Reply:
fixed